vault/token_store.go

```diff
@@ -1904,6 +1908,40 @@ func (ts *TokenStore) handleTidy(ctx context.Context, req *logical.Request, data
 						continue
 					}
 					deletedCountAccessorInvalidToken++
+				default:
+					// Cache the cubbyhole storage key when the token is valid
+					switch {
+					case te.NamespaceID == namespace.RootNamespaceID && !strings.HasPrefix(te.ID, "s."):
```

Comment:
The token's that begin with an `s.` are the service tokens.
A bit of context: The tokens in the root namespace used SHA1 before. We couldn't upgrade to use the SHA2-256 hash here for not being able to distinguish between the tokens that used SHA1 vs the ones that used SHA2-256. Then namespaces came along and we used SHA2-256 for all tokens created in namespaces other than the root namespace (since all the tokens from the beginning would use SHA2-256). After that, service tokens came along which had an `s.` suffix. We ensured that all the tokens that are created all use SHA2-256. Now only the old tokens that were created in the root namespace that were before the `s.` prefix was attached use SHA1 hash. Hope this answers the question.

Reply:
The above comment is the same reason why we are not using the `te.Type`. Not all service tokens use the SHA2-256 hash.

Reply:
The router is also functioning based on the `s.` logic BTW.

Reply:
Yep, sounds good.